smart_contract/src/karma_tiers_test.rs

```diff
@@ -0,0 +1,348 @@
+                            hex::encode(receipt.transaction_hash)
+                        );
+                        println!("   Block number: {:?}", receipt.block_number);
+                        println!("   Gas used: {}", receipt.gas_used);
+                    } else {
+                        println!("Tier update transaction failed");
+                    }
+                }
+                Err(e) => eprintln!("Failed to get receipt: {}", e),
+            },
+            Err(e) => eprintln!("Failed to send tier update transaction: {}", e),
+        }
+    }
+
+    // Test 4: Verify the update by reading the contract again
+    println!("\nVerifying tier update:");
+    println!("=======================");
+
+    match karma_tiers_contract.getTierCount().call().await {
+        Ok(new_tier_count) => {
+            println!("New tier count: {}", new_tier_count);
+
+            if new_tier_count > U256::ZERO {
+                println!("\nUpdated tiers in contract:");
+                println!("============================");
+
+                let tier_count_u8 = new_tier_count.as_limbs()[0] as u8;
+
+                for i in 0..tier_count_u8 {
+                    match karma_tiers_contract.getTierById(i).call().await {
+                        Ok(tier) => {
+                            println!(
+                                "Tier {}: {} ({} - {}) - {} tx/epoch",
+                                i, tier.name, tier.minKarma, tier.maxKarma, tier.txPerEpoch
+                            );
+                        }
+                        Err(e) => {
+                            eprintln!("Failed to get tier {}: {}", i, e);
+                        }
+                    }
+                }
+            }
+        }
+        Err(e) => {
+            eprintln!("Failed to get new tier count: {}", e);
+        }
+    }
+
+    // Test 7: Test tier lookup by karma balance (updated for new tier ranges)
+    println!("\nTesting tier lookup by karma balance:");
+    println!("======================================");
+
+    let test_karma_balances = vec![
+        U256::from(0),    // Below all tiers
+        U256::from(10),   // Basic tier (min)
+        U256::from(25),   // Basic tier (mid)
+        U256::from(49),   // Basic tier (max)
+        U256::from(50),   // Active tier (min)
+        U256::from(75),   // Active tier (mid)
+        U256::from(99),   // Active tier (max)
+        U256::from(100),  // Regular tier (min)
+        U256::from(250),  // Regular tier (mid)
+        U256::from(499),  // Regular tier (max)
+        U256::from(500),  // Power User tier (min)
+        U256::from(750),  // Power User tier (mid)
+        U256::from(999),  // Power User tier (max)
+        U256::from(1000), // S-Tier (min)
+        U256::from(2500), // S-Tier (mid)
+        U256::from(4999), // S-Tier (max)
+        U256::from(5000), // Above all tiers
+    ];
+
+    for karma_balance in test_karma_balances {
```

Comment:
This is not needed here. This should have been tested by the SC team in their unit tests...